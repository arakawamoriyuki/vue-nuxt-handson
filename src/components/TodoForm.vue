<template>
  <div>
    <v-toolbar color="primary" dark flat>
      <v-toolbar-title>Create Todo</v-toolbar-title>
    </v-toolbar>
    <v-card-text>
      <v-form>
        <v-text-field label="title" v-model="title" />
        <v-text-field label="description" v-model="description" />
      </v-form>
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn color="primary" @click="onSubmit({ title, description })">Create</v-btn>
    </v-card-actions>
  </div>
</template>

<script>
export default {
  name: 'todo-form',
  data: () => ({
    title: '',
    description: '',
  }),
  props: {
    onSubmit: Function,
  },
};
</script>
