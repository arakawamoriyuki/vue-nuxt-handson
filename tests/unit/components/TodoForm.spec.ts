import { shallowMount } from '@vue/test-utils';
import TodoForm from '@/components/TodoForm.vue';

describe('TodoForm', () => {
  it('TodoFormが表示される', () => {
    const wrapper = shallowMount(TodoForm);
    expect(wrapper).toMatchSnapshot();
  });
});
