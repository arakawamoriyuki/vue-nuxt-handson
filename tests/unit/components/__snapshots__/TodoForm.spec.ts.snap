// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm TodoFormが表示される 1`] = `
<div>
  <v-toolbar-stub color="primary" dark="true" tag="header" extensionheight="48" flat="true" src="">
    <v-toolbar-title-stub>Create Todo</v-toolbar-title-stub>
  </v-toolbar-stub>
  <v-card-text-stub>
    <v-form-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="title" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
      <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" label="description" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
    </v-form-stub>
  </v-card-text-stub>
  <v-card-actions-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub color="primary" tag="button" activeclass="" type="button">Create</v-btn-stub>
  </v-card-actions-stub>
</div>
`;
